File: Anagram.py
# O(m*n)
class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        res = defaultdict(list)  # mapping charCount

        for s in strs:
            count = [0]*26

            for c in s:
                count[ord(c)-ord("a")] += 1

            res[tuple(count)].append(s)

        return res.values()

File: Anagram.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> map_an;
        for(int i =0; i< strs.size(); i++){
            string s = strs[i];
            sort(strs[i].begin(),strs[i].end());
            map_an[strs[i]].push_back(s);
        }
        for(auto it : map_an ){
            ans.push_back(it.second);
        }
        return ans;
    }
};